src/core/syntax-tree/index.ts

```diff
@@ -1,20 +1,29 @@
+import * as R from 'ramda';
+
 import Logger from 'core/Logger';
-import lexer from 'core/syntax-tree/lexer';
+import lexer, { ILexerResult } from 'core/syntax-tree/lexer';
 import syntaxer, { ISyntaxerResult } from 'core/syntax-tree/syntaxer';
+import { Lexicon } from './lexicon';
 
 export default class SyntaxTree {
-    private result: ISyntaxerResult;
+    protected lexerResult: ILexerResult;
+    protected syntaxerResult: ISyntaxerResult;
```

Comment:
Keeping the intermediary results allows for finer manipulations in the derived syntax tree classes